node/src/effect/requests.rs

```diff
@@ -490,6 +491,14 @@ pub enum ContractRuntimeRequest {
         /// Responder to call with the result.
         responder: Responder<Result<Option<ValidatorWeights>, GetEraValidatorsError>>,
     },
+    /// Performs a step consisting of calculating rewards, slashing and running the auction at the
+    /// end of an era.
+    Step {
+        /// The step request.
+        step_request: StepRequest,
+        /// Responder to call with the result.
+        responder: Responder<Result<StepResult, engine_state::Error>>,
```

Comment:
Topic for later discussion: do we want to allow engine state error to be communicated beyond the boundaries of contract_runtime component?